src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4552,31 +4589,101 @@ private BoundExpression BindCollectionLiteralExpression(CollectionCreationExpres
             {
-                switch (syntax)
+                return syntax switch
                 {
-                    case ExpressionElementSyntax expressionElementSyntax:
-                        return BindValue(expressionElementSyntax.Expression, diagnostics, BindValueKind.RValue);
+                    ExpressionElementSyntax expressionElementSyntax => BindValue(expressionElementSyntax.Expression, diagnostics, BindValueKind.RValue),
+                    DictionaryElementSyntax dictionaryElementSyntax => bindDictionaryElement(dictionaryElementSyntax, diagnostics),
+                    SpreadElementSyntax spreadElementSyntax => bindSpreadElement(spreadElementSyntax, diagnostics),
+                    _ => throw ExceptionUtilities.UnexpectedValue(syntax.Kind())
+                };
+            }
 
-                    case DictionaryElementSyntax dictionaryElementSyntax:
-                        _ = BindValue(dictionaryElementSyntax.KeyExpression, diagnostics, BindValueKind.RValue);
-                        _ = BindValue(dictionaryElementSyntax.ValueExpression, diagnostics, BindValueKind.RValue);
-                        Error(diagnostics, ErrorCode.ERR_CollectionLiteralElementNotImplemented, syntax);
-                        return BadExpression(syntax);
+            BoundExpression bindDictionaryElement(DictionaryElementSyntax syntax, BindingDiagnosticBag diagnostics)
+            {
+                _ = BindValue(syntax.KeyExpression, diagnostics, BindValueKind.RValue);
+                _ = BindValue(syntax.ValueExpression, diagnostics, BindValueKind.RValue);
+                Error(diagnostics, ErrorCode.ERR_CollectionLiteralElementNotImplemented, syntax);
+                return BadExpression(syntax);
+            }
 
-                    case SpreadElementSyntax spreadElementSyntax:
-                        _ = BindValue(spreadElementSyntax.Expression, diagnostics, BindValueKind.RValue);
-                        Error(diagnostics, ErrorCode.ERR_CollectionLiteralElementNotImplemented, syntax);
-                        return BadExpression(syntax);
+            BoundExpression bindSpreadElement(SpreadElementSyntax syntax, BindingDiagnosticBag diagnostics)
+            {
+                var expression = BindRValueWithoutTargetType(syntax.Expression, diagnostics);
+                var builder = new ForEachEnumeratorInfo.Builder();
+                bool hasErrors = !GetEnumeratorInfoAndInferCollectionElementType(syntax, syntax.Expression, ref builder, ref expression, isAsync: false, diagnostics, out _) ||
+                    builder.IsIncomplete;
+                if (hasErrors)
+                {
+                    // PROTOTYPE: Report error.
+                    return BadExpression(syntax);
+                }
 
-                    default:
-                        throw ExceptionUtilities.UnexpectedValue(syntax.Kind());
+                var enumeratorInfo = builder.Build(default);
```

Comment:
consider named param for default.  it's not apparent to me what is means.

Reply:
Added the name, `location:`, although arguably it still not clear what the parameter represents.